### PR TITLE
Use Kontena::Command hooks and new authentication mechanism.

### PR DESCRIPTION
**Pull request by kke**

### CODE REVIEW

File: lib/kontena/machine/azure/cloudinit_master.yml [hunk 1]
```diff
@@ -81,7 +81,8 @@ coreos:
             --link kontena-server-mongo:mongodb \
             -e MONGODB_URI=mongodb://mongodb:27017/kontena_server \
             -e VAULT_KEY=${KONTENA_VAULT_KEY} -e VAULT_IV=${KONTENA_VAULT_IV} \
-            <% if auth_server %>-e AUTH_API_URL=<%= auth_server %><% end %> \
+            <% if server_name %>-e KONTENA_SERVER_NAME=<%= server_name %><% end %> \
```
Comment (jakolehm) [inline lib/kontena/machine/azure/cloudinit_master.yml]: `KONTENA_SERVER_NAME` => `SERVER_NAME`

File: lib/kontena/machine/azure/cloudinit_master.yml [hunk 2]
```diff
@@ -81,7 +81,8 @@ coreos:
...
+            <% if initial_admin_code %>-e KONTENA_INITIAL_ADMIN_CODE=<%= intial_admin_code %><% end %> \
```
Comment (jakolehm) [inline lib/kontena/machine/azure/cloudinit_master.yml]: `KONTENA_INITIAL_ADMIN_CODE`=> `INITIAL_ADMIN_CODE`

**Review by jakolehm — CHANGES_REQUESTED** [review]

**Review by jakolehm — APPROVED** [review]

File: lib/kontena/machine/azure/cloudinit.yml [hunk 3]
```diff
@@ -27,6 +27,16 @@ write_files:
       nameserver 8.8.4.4
 coreos:
   units:
+		- name: etcd2.service
```
Comment (jakolehm) [inline lib/kontena/machine/azure/cloudinit.yml]: something is wrong here?

Comment (kke) [reply lib/kontena/machine/azure/cloudinit.yml]: Perhaps tabs instead of spaces.

**Review by jakolehm — APPROVED** [review]

**Pull request merged**